docs: add a notice of creating bot stack related to frontend WAF

fbukevin opened the pull request:
*Issue #, if available:*
https://github.com/aws-samples/bedrock-claude-chat/issues/633

*Description of changes:*
Just add a notice to README to inform future user notice the bedrock region setting. I think this information in README should be helpful for other users(either the official document or the Caution explicitly).


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

fbukevin committed:
docs: add a notice of creating bot stack related to frontend WAF

Respecting to this isssue:
https://github.com/aws-samples/bedrock-claude-chat/issues/633

I think this information in README should be helpful for other users(either the official document or the Caution explicitly).

fbukevin committed:
docs: add DEPLOYMENT.md for iKala member caution

fbukevin committed:
Update README.md

Add some note of deployment

statefb commented on README.md:
```diff
@@ -64,6 +66,14 @@ By using the [Agent functionality](./docs/AGENT.md), your chatbot can automatica
+## Deployment
+There are two ways to deploy a chatbot:
+
+- Super-easy deployment: This method is designed for using in AWS Cloud Shell or EC2 instance. Or you can use it in your local terminal.
```
This method is designed for the very first user of BrChat (CDK is a little bit difficult way especially for non-tech user).

fbukevin replied:
How about I replace the statement to "This method is designed for the very first user of BrChat"

statefb replied:
lgtm. In addition to that, it'd be great if you could add link to each section.

statefb commented on README.md:
```diff
@@ -64,6 +66,14 @@ By using the [Agent functionality](./docs/AGENT.md), your chatbot can automatica
+There are two ways to deploy a chatbot:
+
+- Super-easy deployment: This method is designed for using in AWS Cloud Shell or EC2 instance. Or you can use it in your local terminal.
+- Using CDK: This method is designed for using in Cloud9. Notice that it requires large amount of space to register container layers. Expand your disk of VM for Cloud9 before you deploy.
```
Not only for Cloud9. Cloud9 will be deprecated, so we don't want to note here. CDK deployment is designed for users who wish to customize e.g. cdk.json.

fbukevin replied:
I got the response from AWS Support that they only mentioned about Cloud 9. I will remove the first part.

statefb commented on DEPLOYMENT.md:
```diff
@@ -0,0 +1,43 @@
+
+## When to bootstrap your environment
```
I think this markdown file should note about FAQ if user failed to deploy like Q&A style. Moreover, this commit contains very specific information e.g. `hnb659fds` but it really depends on each environment. It should be written in generic manner e.g.:

Q. Failed to cdk deploy. How can I solve it?
A. Firstly check bootstrapped correcty by ...ing ....

fbukevin replied:
I think I will remove this part and just create a FAQ either in README.md or other files. (It won't perform in this PR).

statefb commented on README.md:
```diff
@@ -100,6 +110,9 @@ You can specify the following parameters during deployment to enhance security a
 - **--bedrock-region**: Define the region where bedrock is available. (default: us-east-1)
 - **--version**: The version of Bedrock Claude Chat to deploy. (default: latest version in development)
 
+> [!Warning] Currently, the WAF for CloudFront needs to be created in the North America region (us-east-1), so the stacks are separated. https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-wafv2-webacl.html
```
Should this be `Warning`? I think `Note` or `Info` is better

fbukevin replied:
Agree with you. I just think it should be a message at the front before users trying deployment.

fbukevin committed:
Merge branch 'aws-samples:v2' into v2

fbukevin committed:
Update README.md

fbukevin committed:
chore: remove DEPLOYMENT.md since it should be like an FAQ

suzuki-safie committed:
Add dead letter queue to API stack, to limit retries (#660)

* Add dead letter queue to API stack, to limit retries

* fix DLQ name

Change SQS queue DLQ name according to https://qiita.com/tmokmss/items/721a99e9a62499d6d54a. also changed "DLQ" => "Dlq" because not using capitalize-acronym rule.

statefb commented on README.md:
```diff
@@ -64,6 +66,14 @@ By using the [Agent functionality](./docs/AGENT.md), your chatbot can automatica
+## Deployment
+There are two ways to deploy a chatbot:
+
+- Super-easy deployment: This method is designed for the very first user of BrChat.
```
We don't have definition of `BrChat` on this document.

fbukevin replied:
I will remove "of BrChat"

statefb commented on README.md:
```diff
@@ -64,6 +66,14 @@ By using the [Agent functionality](./docs/AGENT.md), your chatbot can automatica
+There are two ways to deploy a chatbot:
+
+- Super-easy deployment: This method is designed for the very first user of BrChat.
+- Using CDK: Notice that it requires large amount of space to register container layers. Expand your disk of VM for Cloud9 before you deploy.
```
Still not fixed?

fbukevin replied:
Ah sorry, I removed the previous one, but omitted the later one.

statefb commented on README.md:
```diff
@@ -64,6 +66,14 @@ By using the [Agent functionality](./docs/AGENT.md), your chatbot can automatica
+- Super-easy deployment: This method is designed for the very first user of BrChat.
+- Using CDK: Notice that it requires large amount of space to register container layers. Expand your disk of VM for Cloud9 before you deploy.
+
+Both method require usig CDK config file in cdk/cdk.js. To use custom bootstrap, please refer to [deploy.md](deploy.md).
```
only cdk deployment requires modification of cdk.json

fbukevin replied:
If you want to use another qualifier, even using bin.sh requires modification cdk.json. 
We had struggle with it very much. Because in deploy.yml, it calls cdk/cdk.json.

statefb replied:
As I mentioned earlier, the target users for super-easy deployment is "the very first user of BrChat". If need to change options which can only be modified via cdk.json (can not change via arguments on bin.sh), it should be an advanced usage so recommend to use deployment by CDK.

fbukevin replied:
Thanks for your confirmation. Actually, that's not work easily. Just as a caution, in the same account, the code of this project sometime will encounters resource confliction when deploying multiple chatbot. We collaborate with AWS support and addressed this issue.

statefb replied:
Understood. Could you clarify what you're ultimately trying to achieve with this PR?

containeers replied:
Please check the pull request #691 for reference to override cdk json and also the custom repo i.e. the forked repo as an argument

fbukevin committed:
chore: Update README.md with some feedback from original owner

fbukevin committed:
chore: replace source clone remote in deploy.yml

statefb commented:
@fbukevin Close due to no communication. thank you
